pkg/controllers/apps/resource/resource.go

```diff
@@ -67,26 +68,30 @@ func NewController(apiResource *metav1.APIResource, clusternetClient *versioned.
 
 	ri := utils.NewResourceInformer(resourceClient, apiResource, cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			resource := obj.(*unstructured.Unstructured)
-			val, ok := resource.GetAnnotations()[known.ObjectOwnedByDescriptionAnnotation]
-			if ok {
-				klog.V(4).Infof("adding %s %q", c.name, klog.KObj(resource))
-				c.enqueue(val)
-			}
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
-			resource := oldObj.(*unstructured.Unstructured)
-			val, ok := resource.GetAnnotations()[known.ObjectOwnedByDescriptionAnnotation]
+			oldresource := oldObj.(*unstructured.Unstructured)
+			newresource := newObj.(*unstructured.Unstructured)
+			oldspec, _, _ := unstructured.NestedMap(oldresource.Object, "spec")
```

Comment:
Some resources like `configmap, secret` neither have `spec` nor `status` .

Reply:
> Some resources like `configmap, secret` neither have `spec` nor `status` .

Thank you for reminding, here I will check the status instead. If it does not have the status, it will not go to the branch that updates the status.

Reply:
``` 
                        newstatus, existed, _ := unstructured.NestedMap(newresource.Object, "status")
			if existed && !reflect.DeepEqual(oldstatus, oldstatus) {
				resAttrs, err = getResourceAttrs(val, ObjectUpdateStatus)
			} else {
				resAttrs, err = getResourceAttrs(val, ObjectUpdateOthers)
			}
```